src/pdf.sandbox.js

```diff
@@ -24,86 +25,124 @@ const TESTING =
   typeof PDFJSDev === "undefined" || PDFJSDev.test("!PRODUCTION || TESTING");
 
 class Sandbox {
-  constructor(module) {
-    this._evalInSandbox = module.cwrap("evalInSandbox", null, [
-      "string",
-      "int",
-    ]);
-    this._dispatchEventName = null;
+  constructor(module, externalNames) {
+    this._commFun = module.cwrap("commFun", null, ["string", "string"]);
     this._module = module;
-    this._alertOnError = 1;
+    this._alertOnError = 0;
+    this._externalNames = externalNames;
   }
 
   create(data) {
-    const sandboxData = JSON.stringify(data);
-    const extra = [
-      "send",
-      "setTimeout",
-      "clearTimeout",
-      "setInterval",
-      "clearInterval",
-      "crackURL",
-    ];
-    const extraStr = extra.join(",");
-    let code = [
-      "exports = Object.create(null);",
-      "module = Object.create(null);",
+    const sandboxData = Sandbox.cloneIn(data);
```

Comment:
This should have been `JSON.stringify`, not `Sandbox.cloneIn`. The result is used to concatenate with a string for JS code, not to pass as a parameter to sandboxed function.